src/main/java/uk/gov/hmcts/reform/blobrouter/tasks/processors/ContainerProcessor.java

```diff
@@ -0,0 +1,50 @@
+package uk.gov.hmcts.reform.blobrouter.tasks.processors;
+
+import com.azure.storage.blob.BlobAsyncClient;
+import com.azure.storage.blob.BlobContainerAsyncClient;
+import com.azure.storage.blob.models.BlobItem;
+import com.azure.storage.blob.specialized.BlobLeaseAsyncClient;
+import com.azure.storage.blob.specialized.BlobLeaseClientBuilder;
+import org.slf4j.Logger;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.slf4j.LoggerFactory.getLogger;
+
+public class ContainerProcessor {
+
+    private static final Logger LOGGER = getLogger(ContainerProcessor.class);
+
+    private final BlobProcessor blobProcessor;
+
+    public ContainerProcessor(BlobProcessor blobProcessor) {
+        this.blobProcessor = blobProcessor;
+    }
+
+    public void process(BlobContainerAsyncClient containerClient) {
+        String containerName = containerClient.getBlobContainerName();
+        AtomicInteger atomInt = new AtomicInteger(0);
+
+        LOGGER.info("Processing container {}", containerName);
+
+        containerClient
+            .listBlobs()
+            .subscribe(
+                blob -> {
+                    processBlobs(containerClient, blob);
+                    atomInt.updateAndGet(operand -> ++operand);
+                },
+                null, // error consumer. can be implemented later
+                () -> LOGGER.info("Finished processing container {}. Blobs found: {}", containerName, atomInt.get())
```

Comment:
Processing a blob may not be completed because of an error or because there was already a lease on it. The number of blobs in the container is useful information, but the number of blobs we actually processed is, in my opinion, the most important piece to include.